platform-sdk/platform-apps/tests/StressTestingTool/src/main/java/com/swirlds/demo/stress/StressTestingToolMain.java

```diff
@@ -0,0 +1,243 @@
+
+    /**
+     * The number of milliseconds of the window period to measure TPS over
+     */
+    private int tps_measure_window_milliseconds = 200;
+
+    /**
+     * The expected TPS for the network
+     */
+    private double expectedTPS = 0;
+
+    /** The constant used to calculate the window size, the higher the expected TPS, the smaller the window */
+    private static final long WINDOW_CALCULATION_CONST = 125000;
+
+    /**
+     * The number of seconds to ramp up the TPS to the expected value
+     */
+    private static final int TPS_RAMP_UP_WINDOW_MILLISECONDS = 20_000;
+
+    /**
+     * The timestamp when the ramp up started
+     */
+    private long rampUpStartTimeMilliSeconds = 0;
+
+    /** App configuration */
+    private StressTestingToolConfig config;
+
+    /**
+     * This is just for debugging: it allows the app to run in Eclipse. If the config.txt exists and lists a particular
+     * SwirldMain class as the one to run, then it can run in Eclipse (with the green triangle icon).
+     *
+     * @param args these are not used
+     */
+    public static void main(final String[] args) {
+        Browser.parseCommandLineArgsAndLaunch(args);
+    }
+
+    public StressTestingToolMain() {
+        transactionGenerator = new StoppableThreadConfiguration<>(getStaticThreadManager())
+                .setComponent("demo")
+                .setThreadName("transaction-generator")
+                .setMaximumRate(50)
+                .setWork(this::generateTransactions)
+                .build();
+    }
+
+    @Override
+    public void updateConfigurationBuilder(@NonNull final ConfigurationBuilder configurationBuilder) {
+        configurationBuilder.withConfigDataType(StressTestingToolConfig.class);
+    }
+
+    @Override
+    public void init(@NonNull final Platform platform, @NonNull final NodeId id) {
+        this.platform = platform;
+        config = platform.getContext().getConfiguration().getConfigData(StressTestingToolConfig.class);
+        expectedTPS = config.transPerSecToCreate()
+                / (double) platform.getAddressBook().getSize();
+
+        // the higher the expected TPS, the smaller the window
+        tps_measure_window_milliseconds = (int) (WINDOW_CALCULATION_CONST / expectedTPS);
+
+        transactionPool = new TransactionPool(config.transPoolSize(), config.bytesPerTrans());
+
+        final Metrics metrics = platform.getContext().getMetrics();
+        transactionSubmitSpeedometer = metrics.getOrCreate(TRAN_SUBMIT_TPS_SPEED_CONFIG);
+    }
+
+    @Override
+    public void run() {
+        final Thread shutdownHook = new ThreadConfiguration(getStaticThreadManager())
+                .setDaemon(false)
+                .setNodeId(platform.getSelfId())
+                .setComponent("app")
+                .setThreadName("demo_log_time_pulse")
+                .setRunnable(() -> {
+                    final Logger logger = LogManager.getLogger(StressTestingToolMain.class);
```

Comment:
logger is already defined in line 55. No need to do it again

Reply:
fixed